src/components/external/DisputeManagement/components/DisputeDetails/DisputeDetails.tsx

```diff
@@ -1,22 +1,50 @@
-import { useCallback, useState } from 'preact/hooks';
-import { DisputeFlowContext, DisputeFlowState } from '../../hooks/useDisputeFlow';
-import { DisputeDetailsContainer } from '../DisputeDetailsContainer/DisputeDetailsContainer';
+import { useMemo } from 'preact/hooks';
+import useCoreContext from '../../../../../core/Context/useCoreContext';
+import { TypographyVariant } from '../../../../internal/Typography/types';
+import Typography from '../../../../internal/Typography/Typography';
 import type { ExternalUIComponentProps } from '../../../../types';
+import { useDisputeFlow } from '../../context/dispute/context';
 import { DisputeManagementProps } from '../../types';
-import { IDisputeDetail } from '../../../../../types/api/models/disputes';
+import { AcceptDisputeFlow } from '../AcceptDisputeFlow/AcceptDisputeFlow';
+import { DefendDisputeFlow } from '../DefendDisputeFlow/DefendDisputeFlow';
+import DisputeData from '../DisputesData/DisputeData';
 
-export const DisputeDetails = (props: ExternalUIComponentProps<DisputeManagementProps>) => {
-    const [flowState, setFlowState] = useState<DisputeFlowState>('details');
-    const goBack = () => setFlowState('details');
-    const [dispute, setDispute] = useState<IDisputeDetail | undefined>();
+export const DisputeDetails = ({
+    id,
+    hideTitle,
+    onDefendDispute,
+    onAcceptDispute,
+    dataCustomization,
+}: ExternalUIComponentProps<DisputeManagementProps>) => {
+    const { flowState, goBack } = useDisputeFlow();
+    const { i18n } = useCoreContext();
 
-    const setDisputeCallback = useCallback((dispute: IDisputeDetail) => {
-        setDispute(dispute);
-    }, []);
-
-    return (
-        <DisputeFlowContext.Provider value={{ flowState, setFlowState, goBack, dispute: dispute, setDispute: setDisputeCallback }}>
-            <DisputeDetailsContainer disputeId={props.id} onAcceptDispute={props.onAcceptDispute} dataCustomization={props.dataCustomization} />
-        </DisputeFlowContext.Provider>
+    const isDefendFlow = useMemo(
+        () => ['defendReasonSelectionView', 'uploadDefenseFilesView', 'defenseSubmitResponseView'].includes(flowState),
+        [flowState]
     );
+
+    if (isDefendFlow) {
+        return <DefendDisputeFlow onDefendDispute={onDefendDispute} />;
+    }
+
+    switch (flowState) {
+        case 'details':
+            return (
+                <>
+                    {flowState === 'details' && !hideTitle && (
+                        <div>
+                            <Typography variant={TypographyVariant.TITLE} medium>
+                                {i18n.get('dispute.management')}
+                            </Typography>
+                        </div>
```

Comment:
Is this wrapper `div` element necessary? You can use the `el={TypographyElement.DIV}` prop directly on the `Typography` component alongside some class names (if needed).

```suggestion
                        <Typography
                            el={TypographyElement.DIV}
                            variant={TypographyVariant.TITLE}
                            medium
                        >
                            {i18n.get('dispute.management')}
                        </Typography>
```